rust/rsc/src/database.rs

```diff
@@ -247,6 +251,38 @@ pub async fn delete_job<T: ConnectionTrait>(
     job::Entity::delete_by_id(job.id).exec(db).await
 }
 
+#[derive(Debug, FromQueryResult)]
+pub struct JobHistoryHash {
+    hash: String,
+}
+
+pub async fn evict_jobs_ttl<T>(db: Arc<T>, ttl: NaiveDateTime) -> Result<usize, DbErr>
+where
+    T: ConnectionTrait + Send + 'static,
+{
+    let hashes = JobHistoryHash::find_by_statement(Statement::from_sql_and_values(
+        DbBackend::Postgres,
+        r#"
+        DELETE FROM job 
+        WHERE created_at <= $1
+        RETURNING hash
+        "#,
+        [ttl.into()],
+    ))
```

Comment:
For my own knowledge, what' s the benefit to direct SQL here rather than the `entity::job::Column::CreatedAt.lte(ttl)` you had previously?

Reply:
The `RETURNING hash` part is required for the job history table and I couldn't get a combo of both "delete many" and "return hash" in the same query. This was the only way (that I could find) to get the correct data out and have it be in only one query